### PR TITLE
Reconfig tx weight fixes

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

- Sends `weight = 0` to contract when the weight (mint rate) is unchanged

- Sends `weight = 1` to contract when the weight is set to 0 by the user

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/v2/V2Project/V2ProjectReconfigureModal/hooks/reconfigureFundingCycle.ts [hunk 1]
```diff
@@ -49,9 +51,26 @@ export const useReconfigureFundingCycle = ({
         }
       : editingFundingCycleMetadata
 
+    const weightBeforeReconfig = fundingCycle?.weight
```
Comment (tomquirk) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/hooks/reconfigureFundingCycle.ts:54]: **suggestion**: 

```suggestion
    const currentFundingCycleWeight = fundingCycle?.weight
```

File: src/components/v2/V2Project/V2ProjectReconfigureModal/hooks/reconfigureFundingCycle.ts [hunk 2]
```diff
@@ -49,9 +51,26 @@ export const useReconfigureFundingCycle = ({
...
+
+    // If weight is 0, we send weight=1 to the contract
+    // If weight has not been modified, we send weight=0 to the contract
+    let weightAfterReconfig = editingFundingCycleData.weight
```
Comment (tomquirk) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/hooks/reconfigureFundingCycle.ts:58]:
```suggestion
    let newFundingCycleWeight = editingFundingCycleData.weight
```

Comment (tomquirk) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/hooks/reconfigureFundingCycle.ts:58]: **suggestion** WDYT about moving this block of code to a function?

Call it `getNewFundingCycleWeight` or something.

**Review by tomquirk — CHANGES_REQUESTED** [review]
@johnnyd-eth minor comment, thanks for picking this up

File: src/components/v2/V2Project/V2ProjectReconfigureModal/hooks/reconfigureFundingCycle.ts (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2ProjectReconfigureModal/hooks/reconfigureFundingCycle.ts:58]: Good call, done

**Comment by johnnyd-eth** [comment]
Thanks @aeolianeth addressed. Given this another good test - confident there are no regressions after the latest updates.

**Review by tomquirk — APPROVED** [review]

**Comment by tomquirk** [comment]
Thank you kind ser! merging

**Pull request merged**